lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py

```diff
@@ -0,0 +1,206 @@
+    - If set to C(vmware_certified), then accept only VIBs that are signed and certified by VMware.
+    - If set to C(vmware_accepted), then accept VIBs that have been accepted by VMware.
+    - If set to C(community), then accept all VIBs, even those that are not signed.
+    choices: [ community, partner, vmware_accepted, vmware_certified ]
+    required: False
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Set acceptance level to community for all ESXi Host in given Cluster
+  vmware_host_acceptance:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: cluster_name
+    acceptance_level: 'community'
+    state: present
+  register: cluster_acceptance_level
+
+- name: Set acceptance level to vmware_accepted for the given ESXi Host
+  vmware_host_acceptance:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    acceptance_level: 'vmware_accepted'
+    state: present
+  register: host_acceptance_level
+
+- name: Get acceptance level from the given ESXi Host
+  vmware_host_acceptance:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    state: list
+  register: host_acceptance_level
+'''
+
+RETURN = r'''
+facts:
+    description:
+    - dict with hostname as key and dict with acceptance level facts, error as value
+    returned: facts
+    type: dict
+    sample: { "facts": { "localhost.localdomain": { "error": "NA", "level": "vmware_certified" }}}
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VMwareAccpetanceManager(PyVmomi):
+    def __init__(self, module):
+        super(VMwareAccpetanceManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.desired_state = self.params.get('state', 'list')
```

Comment:
you should set the default state to "list" at argument_spec, not here.

Reply:
Done.